lib/editing/editor.dart

```diff
@@ -126,7 +126,18 @@ class Position {
 }
 
 abstract class CodeCompleter {
-  Future<List<Completion>> complete(Editor editor);
+  Future<Completions> complete(Editor editor);
+}
+
+class Completions {
```

Comment:
I don't like class names with only a single char change apart. It's too easy to mis-read.

Consider something like CompletionList instead.